File: WrapperCyl/Cyl.py
import math

class Cyl:
    # [init], [default init] and [copy init] put together
    def __init__(self, height: int = 5, rad: int = 5):
        self.height = height
        self.rad = rad
        print('ne proizoshlo copy')

    # destructor
    def __del__(self):
        print('Object {} was destructed'.format(self))

    # instance methods
    def calc_volume(self):
            return math.pi*pow(self.rad, 2)*self.height
        
    def calc_fullSquare(self):
            return 2*math.pi*self.rad*(self.rad+self.height)

if __name__ == '__main__': 
    c = None 
    c = Cyl(5,5)
    print(c)
    a = Cyl(c)
    print(a)
    # print(c)
    # print('kek')
    # d = Cyl(obj = c)
    # print(c.calc_volume())
    # print(c.calc_fullSquare())
    # print(c.height)
    # print(d.height)
    # c.height = 10
    # print(c.height)
    # print(d.height)

File: WrapperCyl/Cyl.cpp
#include "Cyl.h"
#include <cmath>
#include <iostream>
using namespace std;

namespace shapes {

	Cyl::Cyl() {
        this->height = 5;
	    this->rad = 5;
    }

	Cyl::Cyl(double height, double rad) {
		this->height = height;
		this->rad = rad;
	}

	Cyl::Cyl(const Cyl& obj) {
		this->height = obj.height;
		this->rad = obj.rad;
	}

	Cyl::~Cyl() {
		cout << "object was destructed" << endl;
	}


	double Cyl::volume() {
		return 3.14*pow(rad, 2)*height;
	}

	double Cyl::full_Square() {
		return 2 * 3.14*rad*height + 2 * 3 / 14 * pow(rad, 2);
	}

    double Cyl::getHeight() {
        return height;
    }

    void Cyl::setHeight(double height) {
        Cyl::height = height;
    }

	double Cyl::getRad() {
        return rad;
    }

    void Cyl::setRad(double rad) {
        Cyl::rad = rad;
    }
}
